test/org/zaproxy/zap/extension/pscanrulesAlpha/ImageLocationScannerUnitTest.java

```diff
@@ -0,0 +1,147 @@
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.extension.pscanrulesAlpha.ImageLocationScanner;
+
+public class ImageLocationScannerUnitTest extends PassiveScannerTest {
+	private static final int PLUGIN_ID = ImageLocationScanner.PLUGIN_ID;
+	private static final String URI = "https://www.example.com/";
+	private static final String IMAGES_DIR = "test/resources/org/zaproxy/zap/extension/imagelocationscanner";
+
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new ImageLocationScanner();
+	}
+
+	@Test
+	public void passesIfExifLocationDetected() throws HttpMalformedHeaderException, IOException {
+		HttpMessage msg;
+		String fname;
+
+		// Given - image file containing GPS Exif data
+		fname= "exif_gps_01.jpg";
+
+		// When
+		msg = createHttpMessageFromFilename(fname);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		// Then
+		assertEquals(alertsRaised.size(), 1);
+		validateAlert(URI,alertsRaised.get(0));
+		assertThat(alertsRaised.get(0).getEvidence(), containsString("Exif_GPS"));
+	}
+
+	@Test
+	public void passesIfNoIssuesDetected() throws HttpMalformedHeaderException, IOException {
+		HttpMessage msg;
+		String fname;
+
+		// Given - image file with no Exif data
+		fname= "no_alerts_01.jpg";
+
+		// When
+		msg = createHttpMessageFromFilename(fname);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		// Then
+		assertEquals(alertsRaised.size(), 0);
+		
+		// Given
+		// a non-image file, like a text file
+		fname= "README.md";
+
+		// When
+		msg = createHttpMessageFromFilename(fname);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		// Then
+		assertEquals(alertsRaised.size(), 0);
+	}
+
+	@Test
+	public void passesIfPrivacyExposureDetected() throws HttpMalformedHeaderException, IOException {
+		HttpMessage msg;
+		String fname;
+
+		// Given - image with privacy-exposure (embedded camera ownership)
+		fname= "privacy_exposure_01.jpg";
+
+		// When
+		msg = createHttpMessageFromFilename(fname);
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		// Then
+		assertEquals(alertsRaised.size(), 1);
+		assertThat(alertsRaised.get(0).getEvidence(), containsString("Owner Name"));
+	}
+
+	
+	@Test
+	public void testOfScanHttpRequestSend() throws HttpMalformedHeaderException {
+		// the method should do nothing (test just for code coverage)
+		rule.scanHttpRequestSend(null, -1);
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+
+	private static void validateAlert(Alert alert) {
+		validateAlert(URI, alert);
+	}
+
+	private static void validateAlert(String requestUri, Alert alert) {
```

Comment:
This can be merged with previous method, the `requestUri` is always the same (`URI`).